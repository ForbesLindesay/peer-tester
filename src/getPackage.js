function getPackage(entries) {
  for (const entry of entries) {
    if (entry.path === './package.json') {
      try {
        return JSON.parse(entry.content.toString());
      } catch (ex) {
        throw new Error('Syntax error while parsing package.json\n' + ex.message);
      }
    }
  }
  throw new Error('Could not find package.json');
}
export default getPackage;
