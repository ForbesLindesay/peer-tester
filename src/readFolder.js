import lsr from 'lsr';
import {readFile} from 'then-fs';

async function readFolder(dirname) {
  const entries = await lsr(dirname);
  return Promise.all(
    entries.map(async (entry) => {
      if (entry.isFile()) {
        return {
          type: 'file',
          path: entry.path,
          content: await readFile(entry.fullPath),
        };
      } else {
        return {
          type: 'directory',
          path: entry.path,
        };
      }
    }),
  );
}

export default readFolder;
