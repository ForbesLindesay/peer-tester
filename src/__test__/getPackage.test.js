import getPackage from '../getPackage';
import readFolder from '../readFolder';

test('getPackage(readFolder(exampleDir))', async () => {
  const entries = await readFolder(__dirname + '/../../example');
  const pkg = getPackage(entries);
  expect(pkg).toMatchSnapshot();
});

test('getPackage([])', async () => {
  expect(() => getPackage([])).toThrow();
});
