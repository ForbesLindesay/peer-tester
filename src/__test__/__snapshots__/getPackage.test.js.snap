// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getPackage(readFolder(exampleDir)) 1`] = `
Object {
  "name": "example",
  "peerDependencies": Object {
    "promise": "^6.0.0",
  },
  "private": true,
  "scripts": Object {
    "test": "node test",
  },
}
`;
